saw-core-coq/src/Verifier/SAW/Translation/Coq/Term.hs

```diff
@@ -73,44 +100,113 @@ makeLenses ''TranslationReader
 data TranslationState = TranslationState
 
   { _globalDeclarations :: [String]
-  -- ^ Some Cryptol terms seem to capture the name and body of some functions
-  -- they use (whether from the Cryptol prelude, or previously defined in the
-  -- same file).  We want to translate those exactly once, so we need to keep
-  -- track of which ones have already been translated.
+    -- ^ Some Cryptol terms seem to capture the name and body of some functions
+    -- they use (whether from the Cryptol prelude, or previously defined in the
+    -- same file). We want to translate those exactly once, so we need to keep
+    -- track of which ones have already been translated.
 
   , _topLevelDeclarations :: [Coq.Decl]
-  -- ^ Because some terms capture their dependencies, translating one term may
-  -- result in multiple declarations: one for the term itself, but also zero or
-  -- many for its dependencies.  We store all of those in this, so that a caller
-  -- of the translation may retrieve all the declarations needed to translate
-  -- the term.  The translation function itself will return only the declaration
-  -- for the term being translated.
-
-  , _localEnvironment  :: [Coq.Ident]
-  -- ^ The list of Coq identifiers for de Bruijn-indexed local
-  -- variables, innermost (index 0) first.
-
-  , _unavailableIdents :: Set.Set Coq.Ident
-  -- ^ The set of Coq identifiers that are either reserved or already
-  -- in use. To avoid shadowing, fresh identifiers should be chosen to
-  -- be disjoint from this set.
-
-  , _sharedNames :: IntMap.IntMap Coq.Ident
-  -- ^ Index of identifiers for repeated subterms that have been
-  -- lifted out into a let expression.
-
-  , _nextSharedName :: Coq.Ident
-  -- ^ The next available name to be used for a let-bound shared
-  -- sub-expression.
+    -- ^ Because some terms capture their dependencies, translating one term may
+    -- result in multiple declarations: one for the term itself, but also zero
+    -- or many for its dependencies. We store all of those in this, so that a
+    -- caller of the translation may retrieve all the declarations needed to
+    -- translate the term. The translation function itself will return only the
+    -- declaration for the term being translated.
 
   }
   deriving (Show)
 
 makeLenses ''TranslationState
 
+-- | The constraint stating that 'm' can be used for term translation. This
+-- requires that it have reader effects for 'TranslationReader' and state
+-- effects for 'TranslationState'.
 type TermTranslationMonad m =
   TranslationMonad TranslationReader TranslationState m
 
+-- | Get just the 'TranslationReader' component of the reader value
+askTrr :: TermTranslationMonad m => m TranslationReader
+askTrr = otherConfiguration <$> ask
```

Comment:
What does the "`Trr`" in this name stand for? I was half-expecting this to be named "`askTranslationReader`", so I'm not sure where the extra "`r`" comes from.

A similar question for the "`Trr`" in `localTrr`.

Reply:
`askTR` and `localTR` work for me.